doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1676,68 +1615,100 @@ Controlling the proof flow
    .. exn:: @ident is used in conclusion.
       :undocumented:
 
-.. tacn:: generalize @term
-   :name: generalize
-
-   This tactic applies to any goal. It generalizes the conclusion with
-   respect to some term.
+   .. tacn:: specialize_eqs @ident
+      :undocumented:
 
-.. example::
+.. tacn:: generalize @one_term {? {+ @one_term } }
+          generalize @one_term {? at @occs_nums } {? @as_name } {* , @pattern_occs {? @as_name } }
+   :name: generalize; _
 
-   .. coqtop:: reset none
+   For a single :n:`@one_term`, replaces the goal `G` with `forall (x:T), G'`,
+   where :n:`@one_term` is a subterm of `G` of type `T` and `G'` is obtained
+   by replacing all occurrences of :n:`@one_term` with `x` within `G`.  `x` is
+   a fresh variable chosen based on `T`.  Specifying multiple :n:`one_term`\s is
+   equivalent to :n:`generalize @term__n; … ; generalize @term__1`.  (Note they
+   are processed *right to left*.)
 
-      Goal forall x y:nat, 0 <= x + y + y.
-      Proof. intros *.
+   :n:`@occs_nums`
+     Selects which occurrences of :n:`@one_term` to replace (counting from
+     left to right on the expression printed using the :flag:`Printing All` flag)
 
-   .. coqtop:: all abort
+   :n:`@as_name`
+     The name to use for `x` instead of a fresh name.
 
-      Show.
-      generalize (x + y + y).
+   :n:`@pattern_occs`
+     Limits the replacement to terms that match the :n:`@one_term` within :n:`@pattern_occs`.
```

Comment:
Where did you get that from?
I think this is just a case of `generalize` being provided several `@one_term` with optional `@occs_nums` clauses.

Reply:
Yeah, probably wrong, I'll remove it.  BTW, the example just below here looks totally useless.  Perhaps you can suggest a more realistic one?

Reply:
It's actually not completely useless, since it shows that if you have to prove that some complex arithmetic expression is larger than zero, you can just get rid of the expression and just show that any natural number in general is larger than zero.

But to make the example seem even less useless, maybe you are interested in the rest of the proof:

```
induction n.
- easy.
- now constructor.
```

A proof by induction would be much more complex (if not impossible) without the `generalize` step. And this illustrates well what `generalize` is most useful for: generalizing a statement to the right abstraction level before doing a proof by induction.

Reply:
Got it, thanks.